cmd/init.go

```diff
@@ -9,21 +9,27 @@ import (
 	"github.com/supabase/cli/internal/utils"
 )
 
-var initCmd = &cobra.Command{
-	GroupID: groupLocalDev,
-	Use:     "init",
-	Short:   "Initialize a local project",
-	RunE: func(cmd *cobra.Command, args []string) error {
-		fsys := afero.NewOsFs()
-		if err := _init.Run(fsys); err != nil {
-			return err
-		}
+var (
+	createVscodeWorkspace string
 
-		fmt.Println("Finished " + utils.Aqua("supabase init") + ".")
-		return nil
-	},
-}
+	initCmd = &cobra.Command{
+		GroupID: groupLocalDev,
+		Use:     "init",
+		Short:   "Initialize a local project",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			fsys := afero.NewOsFs()
+			if err := _init.Run(fsys, createVscodeWorkspace); err != nil {
+				return err
+			}
+
+			fmt.Println("Finished " + utils.Aqua("supabase init") + ".")
+			return nil
+		},
+	}
+)
 
 func init() {
+	flags := initCmd.Flags()
+	flags.StringVar(&createVscodeWorkspace, "create-vscode-workspace", "", "Generate VS Code workspace.")
```

Comment:
```suggestion
	flags.BoolVar(&createVscodeWorkspace, "with-vscode-workspace", "", "Generate VS Code workspace.")
```

`BoolVar` might be a more suitable choice here.

Reply:
The only issue around this is that a default value is required and as far as I can tell, it can't be `nil`.  It has to be `true`/`false`.  So, I had to add an extra check to see if the user actually set the flag or not.
https://github.com/supabase/cli/pull/1167/files#diff-345cd3beb7d905a8bef99d83e99c62678bfd5101a26d30104ecc9320602072acR21-R23